src/matlab2tikz.m

```diff
@@ -6418,7 +6458,9 @@ function printAll(m2t, env, fid)
     nOpts = size(opts,1);
     c = cell(1,nOpts);
     for k = 1:nOpts
-        if isempty(opts{k,2})
+        if isempty(opts{k,1})
+            c{k} = sprintf('%s', opts{k,2});
```

Comment:
I think I have another way that keeps that data structure cleaner.

In `setColor`, check whether the property name is empty. If it is, do `options = opts_add(options, color)`, otherwise, do what we do now.